services/notification/notifiers/status/base.py

```diff
@@ -336,26 +357,37 @@ async def send_notification(self, comparison: Comparison, payload):
                 "state": state,
                 "message": message,
             }
+
+            all_shas_to_notify = [head_commit_sha] + list(
+                comparison.context.gitlab_extra_shas or set()
+            )
+            if len(all_shas_to_notify) > 1:
+                log.info(
+                    "Notifying multiple SHAs",
+                    extra=dict(all_shas=all_shas_to_notify, commit=head_commit_sha),
+                )
+            all_notify_tasks = [
+                self._send_notification_with_metrics(
+                    repo_service=repository_service,
+                    commit_sha=commitid,
+                    state=state,
+                    title=title,
+                    coverage=(float(head_report.totals.coverage) if head_report else 0),
+                    description=message,
+                    url=url,
+                )
+                for commitid in all_shas_to_notify
+            ]
             try:
-                with metrics.timer(
-                    "worker.services.notifications.notifiers.status.set_commit_status"
-                ):
-                    res = await repository_service.set_commit_status(
-                        commit=head.commitid,
-                        status=state,
-                        context=title,
-                        coverage=(
-                            float(head_report.totals.coverage) if head_report else 0
-                        ),
-                        description=message,
-                        url=url,
-                    )
+                all_results = await asyncio.gather(*all_notify_tasks)
+                res = all_results[0]
```

Comment:
Would we only care of the 1st result here? I guess cause the second+ would be the extra gitlab ones?

Reply:
Exactly. The 1st result matches the notification we were doing before. All others should be as transparent as possible, I suppose